### ISSUE TITLE
Feature request: Run code blocks above

### ISSUE THREAD

**Issue by dalbandea** [gpanders/vim-medieval#4]
I'm interested in literate programming, so it would be nice if one could run code written in different in several code blocks, since it can lead to a cleaner file. For example, one could define some libraries in a code block with a defined `id: library`, and then tell the next code block to run first the block with `id: library`. An example in python would be:
````markdown
<!-- id: library -->
```python
import numpy as np
```  
Some explanation of `numpy` library here. Now example of usage:

<!-- target: array, run-before: library -->
```python
print(np.array([1,2,3])
```

<!-- name: array -->
```
result
```
````
Would it be easy to implement?

**Comment by gpanders** [comment]
I like this idea, and I don't think it would be too difficult to implement. Though I'd likely change `run-before` to `require`, so it would look like this:

````markdown
<!-- name: library -->
```python
import numpy as np
```

<!-- target: array, require: library -->
```python
print(np.array([1,2,3]))
```

<!-- name: array -->
```
result
```
````

I'll take a look at implementing this when I have some time.

**Comment by gpanders** [comment]
Please give the `require` branch a try and let me know if you run into any issues. Also let me know if all looks well, and I'll merge it into master.

**Comment by dalbandea** [comment]
Sorry, I found your plugin yesterday and I didn't have it installed. I'm testing some stuff now but there are some things not working for me (for example, `:EvalBlock` alone is not working, I'll see if I see why and if not I'll open another issue)

So far I can tell you that:
- Your examples of `require` in the docs work if runned with `:EvalBlock!` or `:EvalBlock output` (with `output` being the name of the output block.
- If you misspell the `require` code block, the block runs anyway but without the required code block, of course. This is nice I think, this way I can have a snippet with `<!-- target: blabla, require: -->` leave the `require` field blank if not necessary.
- This is not working for me:

````markdown
<!-- name: library -->
```sh
x=5
```
<!-- target: output, require: library -->
```sh
echo $x
```
````

And sorry again, since I don't know much bout vimscripting and cannot be of much help looking at the code.

**Comment by gpanders** [comment]
Your example wasn't working for me either, and I figured out why. It's a problem that only happens then the very first line is a block label (as in your example). I just pushed a fix for that, so please try again.

**Comment by dalbandea** [comment]
It works now for me with `:EvalBlock output` and `:EvalBlock!`.

**Comment by dalbandea** [comment]
It works correctly for me with `:EvalBlock` as well if I don't have 
```vim
set clipboard=unnamed
```
in my vimrc.

**Comment by gpanders** [comment]
See [my comment](https://github.com/gpanders/vim-medieval/issues/6#issuecomment-786747811) in #6.

**Comment by gpanders** [comment]
Merged in #5.
